tensorflow/core/ops/sparse_ops.cc

```diff
@@ -146,24 +146,33 @@ REGISTER_OP("SparseTensorDenseMatMul")
       TF_RETURN_IF_ERROR(c->WithRank(c->input(0), 2, &unused));  // a_indices
```

Comment:
Shape inference should now be tested in C++. Specifically, this needs a unit test in sparse_ops_test.cc. See examples therein.